src/tracks/playabletrack/wavetrack/ui/WaveTrackAffordanceControls.h

```diff
@@ -11,18 +11,37 @@
 #pragma once
 
 #include <wx/font.h>
+#include <wx/event.h>
+
 #include "../../../ui/CommonTrackPanelCell.h"
+#include "../../../ui/TextEditDelegate.h"
+
+
+struct TrackListEvent;
 
 class AffordanceHandle;
 class WaveClip;
+class TextEditHelper;
 class TrackPanelResizeHandle;
+class WaveClipTitleEditHandle;
 class WaveTrackAffordanceHandle;
+class TrackList;
 
-class AUDACITY_DLL_API WaveTrackAffordanceControls : public CommonTrackCell
+//Handles clip movement, selection, navigation and
+//allow name change
+class AUDACITY_DLL_API WaveTrackAffordanceControls : 
+    public CommonTrackCell,
+    public TextEditDelegate,
+    public wxEvtHandler,
+    public std::enable_shared_from_this<WaveTrackAffordanceControls>
 {
     std::weak_ptr<WaveClip> mFocusClip;
     std::weak_ptr<WaveTrackAffordanceHandle> mAffordanceHandle;
     std::weak_ptr<TrackPanelResizeHandle> mResizeHandle;
+    std::weak_ptr<WaveClipTitleEditHandle> mTitleEditHandle;
+
+    std::shared_ptr<TextEditHelper> mTextEditHelper;
```

Comment:
Now I understand why TextEditHelper has a weak pointer to its delegate.  If it were shared, we would have a cycle of shared pointers which is a bad thing.

The helper is owned by its delegate and not the opposite.

Perhaps comments in those text edit classes should clarify this point of lifetime management?